### ISSUE TITLE
Support for eqnarray

### ISSUE THREAD

**Issue by snorky22** [KaTeX/KaTeX#3643]
> Before requesting a feature, please search for [existing issues](https://github.com/KaTeX/KaTeX/issues).

**Is your feature request related to a problem? Please describe.**
The eqnarray environment is a classic; it is described in the original LaTeX book by Leslie Lamport. Many people I know use it because it is a core component of LaTeX. Unfortunately, it is not supported by KaTeX (I have tried).
I**Describe the solution you'd like:**
Support for the eqnarray environment (both numbered and un-numbered).
**Link to or name of a (La)TeX package that provides the same feature:**
The eqnarray environment is included in the core LaTeX code. No package needed.
**Describe alternatives you've considered:**
It has been suggested in the discussions to use the alignedat environment as a substitute.
I have tried it and the alignedat environment is definitively not a substitute for the eqnarray environment .
In an eqnarray, the first part is flushed right, the second part is centered and the third part is flushed left.
 You do not have that at all in alignedat. 
**Additional context:**
Mathjax does support eqnarray.

**Comment by edemaine** [comment]
I agree that we should support `eqnarray`. This is possible now that we support top-level environments (currently only AMS environments).

For now, the following workaround is probably reasonable:
```latex
$$\begin{darray}{rcl} ... \end{darray}$$
```
But I would guess that this doesn't get the spacing right.
